src/Symfony/Component/ErrorHandler/FatalErrorHandler/FatalErrorHandlerInterface.php

```diff
@@ -23,9 +23,7 @@ interface FatalErrorHandlerInterface
     /**
      * Attempts to convert an error into an exception.
      *
-     * @param array $error An array as returned by error_get_last()
-     *
-     * @return FatalErrorException|null A FatalErrorException instance if the class is able to convert the error, null otherwise
+     * @return FatalErrorInterface|null A FatalErrorInterface instance if the class is able to improve the error, null otherwise
      */
-    public function handleError(array $error, FatalErrorException $exception);
+    public function handleError(FatalErrorInterface $fatalError): ?FatalErrorInterface;
```

Comment:
I think it would be good to get rid of `FatalErrorHandlerInterface`. Can't we handle this with the ErrorEnhancers as well? Maybe that requires to change the type hints in ErrorEnhancers from \Error to \Throwable. But that seems fine anyway because the ErrorEnhancers could also be used for Exceptions in theory.

Reply:
Yes!